packages/protocol/test/common/onboarding.ts

```diff
@@ -0,0 +1,424 @@
+const Attestations: AttestationsTestContract = artifacts.require('AttestationsTest')
+const MTW: MetaTransactionWalletContract = artifacts.require('MetaTransactionWallet')
+const Random: MockRandomContract = artifacts.require('MockRandom')
+contract('Komenci Onboarding', (_accounts: string[]) => {
+  let accounts: AccountsInstance
+  let stableToken: StableTokenInstance
+  let mtw: MetaTransactionWalletInstance
+  let random: MockRandomInstance
+  let registry: RegistryInstance
+  let relayermtw: MetaTransactionWalletInstance
+  let transactions: any[]
+  const attestationExpiryBlocks = (60 * 60) / 5
+  const selectIssuersWaitBlocks = 4
+  const maxAttestations = 20
+  const user: string = _accounts[1]
+  // Random hex strings
+  const dek = '0x02f2f48ee19680706196e2e339e5da3491186e0c4c5030670656b0e01611111111'
+  const identifier = '0x02f2f48ee19680706196e2e339e5da3491186e0c4c5030670656b0e016111111'
+  const numAttestations = 3
+  // TODO(asa): Clean this up
+  // const perAttestationFee = web3.utils.toWei(numAttestations * 0.05, 'ether').toString()
+  const perAttestationFee = '50000000000000000'
+  const attestationFee = '150000000000000000'
+  before(async () => {
+    stableToken = await getDeployedProxiedContract('StableToken', artifacts)
+    // The komenci version is set as a test contract because we are not using it with a proxy.
+    relayermtw = await MTW.new(true)
+    await relayermtw.initialize(_accounts[0])
+
+    // Set up the required mocks that will allow verification to work with ganache.
+    registry = await getDeployedProxiedContract('Registry', artifacts)
+    random = await Random.new()
+    accounts = await Accounts.new()
+    await accounts.initialize(registry.address)
+    // Take ownership of the registry contract to point it to the mocks.
+    await assumeOwnership(['Registry'], _accounts[0])
+    await registry.setAddressFor(CeloContractName.Random, random.address)
+    await registry.setAddressFor(CeloContractName.Accounts, accounts.address)
+    const Validators: MockValidatorsContract = artifacts.require('MockValidators')
+    const mockValidators = await Validators.new()
+    await registry.setAddressFor(CeloContractName.Validators, mockValidators.address)
+
+    await Promise.all(
+      _accounts.map(async (account) => {
+        await accounts.createAccount({ from: account })
+        await unlockAndAuthorizeKey(
+          KeyOffsets.VALIDATING_KEY_OFFSET,
+          accounts.authorizeValidatorSigner,
+          account,
+          _accounts
+        )
+        await unlockAndAuthorizeKey(
+          KeyOffsets.ATTESTING_KEY_OFFSET,
+          accounts.authorizeAttestationSigner,
+          account,
+          _accounts
+        )
+      })
+    )
+  })
+
+  describe('With AttestationsV1', () => {
+    let attestations: AttestationsTestInstance
+    before(async () => {
+      // Attestations uses some precompiles that are not implemented in ganache.
+      // TODO: Put a proxy in front of it.
```

Comment:
Is this something you want to get done before this PR is merged?

Reply:
Not really, it means our gas numbers will be off slightly but they'll be equally off for all approaches and only very slightly so I don't think it matters much.